### ISSUE TITLE
Trouble using SVG images in the newest form of manim

### ISSUE THREAD

**Issue by luketimmerman314** [3b1b/manim#922]
Hello, I am having trouble getting an .svg image to work in the newest version of manim. Any help would be appreciated. I have attached a screen shot of the code I am using, and the error messages that I am getting. In making my own code, I tried to mimic the PiCreatures code as close as possible to get it to run but to no avail.

![Screen Shot 2020-03-17 at 3 41 03 PM](https://user-images.githubusercontent.com/62107786/76899643-f6cf7c80-6865-11ea-87ac-aaeebb6d7539.png)
![Screen Shot 2020-03-17 at 3 41 21 PM](https://user-images.githubusercontent.com/62107786/76899650-fa630380-6865-11ea-92e5-ed16b4a34b80.png)
![Screen Shot 2020-03-17 at 3 41 42 PM](https://user-images.githubusercontent.com/62107786/76899659-fd5df400-6865-11ea-8759-2c515fc4a91a.png)
![Screen Shot 2020-03-17 at 3 41 58 PM](https://user-images.githubusercontent.com/62107786/76899663-ff27b780-6865-11ea-8376-c738552b8231.png)

**Comment by mental32** [comment]
This doesn't appear to be an issue with the library, at a guess your missing `FILE_DIR` appears to be either `PSI_CREATURE_DIR` or `MEDIA_DIR`.

Also for future refrence it's better to attach a link to a [gist](https://gist.github.com/) or [hastebin](https://hastebin.com) with the relevant code and tracebacks

**Comment by luketimmerman314** [comment]
So how would I remedy this situation? How do I create a PSI_CREATURE_DIR or a MEDIA_DIR?

**Comment by luketimmerman314** [comment]
Also, where is the image directory?

**Comment by mental32** [comment]
> So how would I remedy this situation? How do I create a PSI_CREATURE_DIR or a MEDIA_DIR?

Both of those names are defined already at the top of the file so it's as simple as replacing the undefined name with either one of those.

Also for getting started with Python I'd recommend the [official tutorial](https://docs.python.org/3/tutorial/) or the [realpython blog](https://realpython.com/) and for asking questions there's [r/learnpython](https://www.reddit.com/r/learnpython/), one of the [many](https://discordapp.com/invite/python) [discord](https://discord.gg/RAKc3HF) [guilds](https://discordapp.com/invite/code) or the #python channel on the [freenode IRC](https://freenode.net/).

**Comment by luketimmerman314** [comment]
Ok, so I changed the FILE_DIR to PSI_CREATURE_DIR, and that got rid of the FILE_DIR Not Defined Error. Thank you for that.

But it still says 
OSError: No file matching assets/PsiCreatures_happy.svg in image directory

Does this have something to do with where I have placed the svg file? Currently, I have placed the svg file in manim/manimlib/files, as this is where the PiCreatures_plain.svg also resides. I tried moving the svg to the assets file to see if that would work, but it still returns the same error message. Any advice for this?

**Comment by hanwenzhu** [comment]
`FILE_DIR` was defined way back to be os.path.join(".", "files") and is removed in commit caa4577cd161e5c8bd4f2f57b9d6c3295672d2a3. Add this line to the top of your script
```py
FILE_DIR = os.path.join(".", "files")
```
after the `import *` if your SVG files are placed in ./files

**Comment by luketimmerman314** [comment]
Ok, so I moved the file PsiCreatures_happy.svg back to the files folder in the library and then added the code you provided. However, it is still giving me this error when I am trying to run it in Terminal:

Media will be written to ./media/. You can change this behavior with the --media_dir flag.
conversation.py:39: UserWarning: No PsiCreatures design with mode happy
  (self.file_name_prefix, mode))



Traceback (most recent call last):
  File "conversation.py", line 36, in __init__
    SVGMobject.__init__(self, file_name=svg_file, **kwargs)
  File "/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py", line 45, in __init__
    self.ensure_valid_file()
  File "/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py", line 63, in ensure_valid_file
    self.file_name)
OSError: No file matching files/PsiCreatures_happy.svg in image directory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/luketimmerman/Animation/manim/manimlib/extract_scene.py", line 155, in main
    scene = SceneClass(**scene_kwargs)
  File "/Users/luketimmerman/Animation/manim/manimlib/scene/scene.py", line 53, in __init__
    self.construct()
  File "conversation.py", line 70, in construct
    psi_creature = PsiCreature()
  File "conversation.py", line 44, in __init__
    SVGMobject.__init__(self, mode="happy", file_name=svg_file, **kwargs)
  File "/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py", line 45, in __init__
    self.ensure_valid_file()
  File "/Users/luketimmerman/Animation/manim/manimlib/mobject/svg/svg_mobject.py", line 63, in ensure_valid_file
    self.file_name)
OSError: No file matching assets/PsiCreatures_happy.svg in image directory

Any ideas? Does it have something to do with the fact that it is trying to work with the assets folder? Because this folder is no longer where those svg images are stored.

Thanks!

**Comment by hanwenzhu** [comment]
The `files` folder is meant to be in the directory that you called your python program (presumably the same directory as `conversation.py`

**Comment by luketimmerman314** [comment]
Ok, thank you so much! That got it to play, but now there is a different issue: it looks nothing like the image that I made using the InkScape program. Any suggestions on how to fix that? I tried using it with Grant's Pi Creatures, and that worked perfectly, so it must be something with my file or code, maybe?

Thanks!

**Comment by hanwenzhu** [comment]
I'm not sure, but my guess is that it's a SVG file problem, since if I remember correctly manim uses its own SVG parsing code. This works fine on LaTeX-generated SVG but most likely not on any SVG.

**Comment by luketimmerman314** [comment]
Ok thanks for the clarification. Is there somewhere you could point me that would show me how to generate my svg file using LaTex? Or perhaps somehow convert my .svg to be compatible with the manim code?

Thanks again for all of your help! I really appreciate it!

**Comment by hanwenzhu** [comment]
The .tex file generates a .dvi file using the `latex` command, which is used to generate the .svg file using `dvisvgm`. This is in [tex_file_writing.py](https://github.com/3b1b/manim/blob/master/manimlib/utils/tex_file_writing.py). The SVG parsing part is at the definition of SVGMobject [here](https://github.com/3b1b/manim/blob/master/manimlib/mobject/svg/svg_mobject.py#L76). It only handles `<defs>`, `<g>/<svg>/<symbol>`, `<path>`, `<use>`, `<rect>`, `<circle>`, `<ellipse>`, and `<polygon>/<polyline>`.

**Comment by luketimmerman314** [comment]
All that my PsiCreature is built from are paths and ellipses, so I'm not sure if that's the problem. So I found the image that Grant uses (PiCreatures_plain.svg) and opened it with just the standard TextEdit Program. It looks as if Grant used Adobe Illustrator to construct the image, and then used an SVG Export plug In to export the file. Do you think that if I rebuilt the image using Illustrator instead of InkScape that it would render correctly? For reference, I have attached the code I received from PiCreatures_plain.svg

<?xml version="1.0" encoding="utf-8"?>

<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 480 480" style="enable-background:new 0 0 480 480;" xml:space="preserve">
<style type="text/css">
	.st0{fill:#FFFFFF;}
	.st1{fill:#0C7F99;}
</style>
<path class="st0" d="M251.6,114.7c0,4.2-1.4,8.2-3.7,11.3l-30.4,0.3c-2.6-3.1-4-7.3-4-11.6c0-10.5,8.5-19,19-19
	C243.1,95.6,251.6,104.2,251.6,114.7z"/>
<path class="st0" d="M306.6,114.7c0,4.2-1.4,8.2-3.7,11.3l-30.4,0.3c-2.6-3.1-4-7.3-4-11.6c0-10.5,8.5-19,19-19
	C298.1,95.6,306.6,104.2,306.6,114.7z"/>
<path d="M238,99.8c-1.3,0.1-2.3,1.1-2.3,2.4c0,1.3,1.1,2.4,2.4,2.4c1.3,0,2.4-1.1,2.4-2.4c0-1.3-1.1-2.4-2.4-2.4
	C238.1,99.8,238,99.8,238,99.8l2.9-1c4.3,0,7.7,3.5,7.7,7.7s-3.5,7.7-7.7,7.7s-7.7-3.5-7.7-7.7s3.5-7.7,7.7-7.7"/>
<path d="M293,99.8c-1.3,0.1-2.3,1.1-2.3,2.4c0,1.3,1.1,2.4,2.4,2.4s2.4-1.1,2.4-2.4c0-1.3-1.1-2.4-2.4-2.4
	C293.1,99.8,293,99.8,293,99.8l2.9-1c4.3,0,7.7,3.5,7.7,7.7s-3.5,7.7-7.7,7.7s-7.7-3.5-7.7-7.7s3.5-7.7,7.7-7.7"/>
<path class="st1" d="M228.4,157.1h64.4c-15.4,63.8-25.1,105.4-25.1,151c0,8,0,69.5,23.4,69.5c12,0,22.2-10.8,22.2-20.5
	c0-2.8,0-4-4-12.5c-15.4-39.3-15.4-88.3-15.4-92.3c0-3.4,0-43.9,12-95.1h63.8c7.4,0,26.2,0,26.2-18.2c0-12.5-10.8-12.5-21.1-12.5
	H187.4c-13.1,0-32.5,0-58.7,27.9c-14.8,16.5-33,46.7-33,50.1s2.8,4.6,6.3,4.6c4,0,4.6-1.7,7.4-5.1c29.6-46.7,59.2-46.7,73.5-46.7
	h32.5c-12.5,42.7-26.8,92.3-73.5,192c-4.6,9.1-4.6,10.3-4.6,13.7c0,12,10.3,14.8,15.4,14.8c16.5,0,21.1-14.8,27.9-38.7
	c9.1-29.1,9.1-30.2,14.8-53L228.4,157.1"/>
<path d="M246.3,149.1c5.5,0,1.9,0.1,13.5,0c10.3-0.1,12.4-0.1,15.7-0.2c0,0-0.3-3.1-0.3-3.1c-3.7,0.1-9.3,0.2-14.7,0.2
	c-3.9,0-8.5,0-14.6,0"/>
</svg>

Here is my code that I get for my PsiCreatures_happy.svg also:

<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg
   xmlns:dc="http://purl.org/dc/elements/1.1/"
   xmlns:cc="http://creativecommons.org/ns#"
   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
   xmlns:svg="http://www.w3.org/2000/svg"
   xmlns="http://www.w3.org/2000/svg"
   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
   width="210mm"
   height="297mm"
   viewBox="0 0 210 297"
   version="1.1"
   id="svg8"
   inkscape:version="1.0beta2 (2b71d25, 2019-12-03)"
   sodipodi:docname="psi_creature_happy.svg">
  <defs
     id="defs2" />
  <sodipodi:namedview
     id="base"
     pagecolor="#ffffff"
     bordercolor="#666666"
     borderopacity="1.0"
     inkscape:pageopacity="0.0"
     inkscape:pageshadow="2"
     inkscape:zoom="0.63261527"
     inkscape:cx="213.73446"
     inkscape:cy="398.71679"
     inkscape:document-units="mm"
     inkscape:current-layer="layer1"
     inkscape:document-rotation="0"
     showgrid="false"
     inkscape:window-width="1440"
     inkscape:window-height="805"
     inkscape:window-x="0"
     inkscape:window-y="23"
     inkscape:window-maximized="0" />
  <metadata
     id="metadata5">
    <rdf:RDF>
      <cc:Work
         rdf:about="">
        <dc:format>image/svg+xml</dc:format>
        <dc:type
           rdf:resource="http://purl.org/dc/dcmitype/StillImage" />
        <dc:title />
      </cc:Work>
    </rdf:RDF>
  </metadata>
  <g
     inkscape:label="Layer 1"
     inkscape:groupmode="layer"
     id="layer1">
    <path
       style="fill:#05d5fa;stroke:#000000;stroke-width:0.965;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1;fill-opacity:1;stroke-miterlimit:4;stroke-dasharray:none"
       d="m 86.543101,117.34948 -6.674715,-0.28525 -9.470109,-1.14097 -6.50357,-1.48327 -6.560616,-2.6813 -4.792104,-2.39605 -3.47998,-2.6813 -3.137686,-2.62424 -2.90949,-3.87933 L 41.189267,96.925991 38.507971,89.053248 37.823386,84.831636 37.1388,77.301187 36.796506,70.113032 36.62536,64.065856 36.454212,58.931457 35.65553,53.797062 34.457502,51.115766 32.746037,49.461351 31.20572,48.49152 28.296227,47.863982 h -0.627536 l 0.02889,-3.672229 2.253062,0.07815 3.765222,0.285244 2.795394,0.627536 2.567199,0.51344 3.594077,1.312124 2.738344,1.597367 2.681296,2.4531 2.396053,3.251785 2.224905,4.221614 0.855732,1.825565 1.140975,3.93637 0.798685,5.533736 0.45639,4.906201 0.228196,4.278664 0.171148,3.47998 0.285245,3.93637 0.741635,4.050469 0.741635,3.708175 1.312121,3.651126 1.54032,2.453101 1.711465,2.11081 1.93966,1.82556 2.05376,1.48327 2.90949,1.76852 2.738345,1.02688 3.137686,1.02687 3.993419,0.68459 3.02359,0.2282 3.93637,0.11409 0.114099,-46.437761 -0.399343,-3.87932 -0.399341,-3.308834 -1.198026,-2.567197 -2.167858,-2.224905 -2.96654,-1.711465 -3.308831,-0.684586 -3.308835,-0.399343 -3.87932,-0.285245 -1.198026,0.114099 -0.05705,-3.651126 55.679681,0.114097 0.17114,3.42293 -2.73834,0.114099 -3.19474,0.114096 -3.25178,0.570489 -3.19474,1.02688 -2.85244,1.768515 -1.65442,1.654416 -0.96983,1.825564 -0.68458,2.738345 -0.28525,3.194735 -0.22819,4.335711 -0.0571,5.248494 v 6.560617 l 0.0571,32.232605 h 3.99342 l 3.70817,-0.3423 3.99342,-0.74163 3.30883,-1.08393 3.53703,-1.65442 2.339,-1.65441 2.51015,-2.28196 2.4531,-2.96654 1.14098,-2.167856 1.59737,-4.050469 0.74163,-4.050466 0.57049,-5.077349 0.39934,-4.96325 0.3423,-6.674715 0.68458,-4.506857 0.62754,-4.164565 1.65442,-4.96325 1.25507,-2.852444 1.54032,-2.738345 1.99671,-2.624246 3.08064,-2.4531 3.13768,-1.711465 5.1344,-1.825564 3.47998,-0.399341 4.16457,-0.342294 1.59736,0.114099 -0.057,3.36588 -2.5672,0.51344 -2.51015,1.255075 -1.76852,2.053759 -0.79868,1.99671 -0.74164,3.137686 -0.28524,2.738345 -0.2282,3.194735 -0.057,2.624246 -0.57049,13.691724 -0.28525,3.651126 -0.62753,2.738345 -0.62754,4.107516 -1.65442,5.761934 -1.31212,2.795395 -1.71147,2.852441 -2.2249,2.62425 -2.5672,2.6813 -2.39605,1.76851 -3.25179,2.16786 -3.70817,1.93966 -4.56391,1.65441 -3.59408,1.08393 -3.82227,0.85573 -5.64783,1.02688 -5.41964,0.45639 -4.10752,0.39935 -1.82556,-0.17115 -0.17115,16.14482 0.34229,20.82283 0.3423,3.53703 0.51344,3.02359 0.85573,1.59737 1.48327,1.65441 1.93966,1.31213 1.71146,0.91278 1.99671,0.62753 2.51015,0.62754 3.30884,0.28525 3.99342,0.057 v 3.59407 l -55.736731,-0.11409 0.05705,-3.47998 4.735054,-0.17115 3.936371,-0.51344 3.365881,-1.08393 2.339004,-1.14098 1.312124,-0.96983 0.96983,-1.31212 0.96983,-1.48327 0.627536,-2.22491 0.57049,-3.99341 v -4.79211 l 0.171145,-14.88975 z"
       id="path24"
       inkscape:connector-curvature="0" />
    <path
       style="fill:none;stroke:#000000;stroke-width:0.748;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1;stroke-miterlimit:4;stroke-dasharray:none"
       d="m 100.74351,45.639074 h 22.5961 c 0,0 -1.32918,-13.54538 -11.96264,-14.28422 -10.63346,-0.73884 -10.63346,14.28422 -10.63346,14.28422 z"
       id="path50"
       inkscape:connector-curvature="0" />
    <ellipse
       style="fill:#050000;fill-opacity:1;fill-rule:evenodd;stroke-width:0.145492"
       id="path52"
       cx="111.24973"
       cy="39.960098"
       rx="3.1074717"
       ry="2.957778" />
    <path
       inkscape:connector-curvature="0"
       id="path50-5"
       d="m 67.636488,45.513256 h 22.596103 c 0,0 -1.32918,-13.54538 -11.96264,-14.28422 -10.633463,-0.73884 -10.633463,14.28422 -10.633463,14.28422 z"
       style="fill:none;stroke:#000000;stroke-width:0.848;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1;stroke-miterlimit:4;stroke-dasharray:none" />
    <ellipse
       ry="2.957778"
       rx="3.1074717"
       cy="39.477459"
       cx="78.607201"
       id="path52-8"
       style="fill:#050000;fill-opacity:1;fill-rule:evenodd;stroke-width:0.145492" />
    <path
       style="fill:#050000;fill-opacity:1;stroke:#000000;stroke-width:0.108385px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1"
       d="m 75.959759,61.477064 c 13.457701,5.68591 26.622331,4.339439 39.675031,0.305194 -4.21382,3.690331 -9.14664,6.997189 -19.379726,7.477228 C 85.414893,68.929287 79.496059,65.864988 75.959759,61.477064 Z"
       id="path907"
       inkscape:connector-curvature="0"
       sodipodi:nodetypes="cccc" />
  </g>
</svg>

**Comment by luketimmerman314** [comment]
Sorry, it looks all of the code wasn't posted. Let me try again. If it doesn't post this time, my question about building the file in Adobe Illustrator still stands.

<?xml version="1.0" encoding="utf-8"?> 

<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 480 480" style="enable-background:new 0 0 480 480;" xml:space="preserve">
<style type="text/css">
	.st0{fill:#FFFFFF;}
	.st1{fill:#0C7F99;}
</style>
<path class="st0" d="M251.6,114.7c0,4.2-1.4,8.2-3.7,11.3l-30.4,0.3c-2.6-3.1-4-7.3-4-11.6c0-10.5,8.5-19,19-19
	C243.1,95.6,251.6,104.2,251.6,114.7z"/>
<path class="st0" d="M306.6,114.7c0,4.2-1.4,8.2-3.7,11.3l-30.4,0.3c-2.6-3.1-4-7.3-4-11.6c0-10.5,8.5-19,19-19
	C298.1,95.6,306.6,104.2,306.6,114.7z"/>
<path d="M238,99.8c-1.3,0.1-2.3,1.1-2.3,2.4c0,1.3,1.1,2.4,2.4,2.4c1.3,0,2.4-1.1,2.4-2.4c0-1.3-1.1-2.4-2.4-2.4
	C238.1,99.8,238,99.8,238,99.8l2.9-1c4.3,0,7.7,3.5,7.7,7.7s-3.5,7.7-7.7,7.7s-7.7-3.5-7.7-7.7s3.5-7.7,7.7-7.7"/>
<path d="M293,99.8c-1.3,0.1-2.3,1.1-2.3,2.4c0,1.3,1.1,2.4,2.4,2.4s2.4-1.1,2.4-2.4c0-1.3-1.1-2.4-2.4-2.4
	C293.1,99.8,293,99.8,293,99.8l2.9-1c4.3,0,7.7,3.5,7.7,7.7s-3.5,7.7-7.7,7.7s-7.7-3.5-7.7-7.7s3.5-7.7,7.7-7.7"/>
<path class="st1" d="M228.4,157.1h64.4c-15.4,63.8-25.1,105.4-25.1,151c0,8,0,69.5,23.4,69.5c12,0,22.2-10.8,22.2-20.5
	c0-2.8,0-4-4-12.5c-15.4-39.3-15.4-88.3-15.4-92.3c0-3.4,0-43.9,12-95.1h63.8c7.4,0,26.2,0,26.2-18.2c0-12.5-10.8-12.5-21.1-12.5
	H187.4c-13.1,0-32.5,0-58.7,27.9c-14.8,16.5-33,46.7-33,50.1s2.8,4.6,6.3,4.6c4,0,4.6-1.7,7.4-5.1c29.6-46.7,59.2-46.7,73.5-46.7
	h32.5c-12.5,42.7-26.8,92.3-73.5,192c-4.6,9.1-4.6,10.3-4.6,13.7c0,12,10.3,14.8,15.4,14.8c16.5,0,21.1-14.8,27.9-38.7
	c9.1-29.1,9.1-30.2,14.8-53L228.4,157.1"/>
<path d="M246.3,149.1c5.5,0,1.9,0.1,13.5,0c10.3-0.1,12.4-0.1,15.7-0.2c0,0-0.3-3.1-0.3-3.1c-3.7,0.1-9.3,0.2-14.7,0.2
	c-3.9,0-8.5,0-14.6,0"/>
</svg>

**Comment by hanwenzhu** [comment]
I'm not sure of that, but if PiCreatures_plain works on your `PsiCreature` class, I guess using Adobe Illustrator would make it work. If you want you can upload the SVG file to pastebin or GitHub gist.

**Comment by luketimmerman314** [comment]
Ok, thanks! I'll try it and see!

**Comment by soulimane-mammar** [comment]
I solved this problem by creating an assets/PiCreature folder in the current directory and put all svg files in it

**Comment by soulimane-mammar** [comment]
All but Picreatures are missing except for plain
